### PR TITLE
feat(GHA): add gha for API

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

This pull request introduces a new GitHub Actions workflow to automate the build and push process for API containers. The workflow is triggered by pushes to the master branch and published releases, and it includes steps for setting up the environment, logging into DockerHub, and building and pushing the container images.

Key changes include:

* Added a new workflow configuration file `.github/workflows/api-build-lint-push-containers.yml` to automate the build and push process for API containers.
* Configured the workflow to trigger on pushes to the master branch and published releases.
* Set up environment variables for DockerHub repository, image names, and Python version.
* Included steps for checking out the code, setting up Python, logging into DockerHub, and building and pushing container images.
### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/api-build-lint-push-containers.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        working-directory: ${{ env.WORKING_DIRECTORY }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
```
Comment (jfagoagas) [inline .github/workflows/api-build-lint-push-containers.yml]:
```suggestion
```

File: .github/workflows/api-build-lint-push-containers.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  # Tags
+  LATEST_TAG: latest
+  RELEASE_TAG: ${{ github.event.release.tag_name }}
+
+  # Python configuration
+  PYTHON_VERSION: 3.12
+
+  WORKING_DIRECTORY: ./api
+  DOCKERFILE_PATH: ./api/Dockerfile
+
```
Comment (jfagoagas) [inline .github/workflows/api-build-lint-push-containers.yml]:
```suggestion
```

File: .github/workflows/api-build-lint-push-containers.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  # Container Registries
+  PROWLERCLOUD_DOCKERHUB_REPOSITORY: prowlercloud
+  PROWLERCLOUD_DOCKERHUB_IMAGE: prowler-api
+
+jobs:
+  # Build Prowler OSS container
+  container-build-push:
+    runs-on: ubuntu-latest
+    env:
+      POETRY_VIRTUALENVS_CREATE: "false"
```
Comment (jfagoagas) [inline .github/workflows/api-build-lint-push-containers.yml]:
```suggestion
```

**Review by drewkerrigan — DISMISSED** [review]
Thanks @jfagoagas!

File: .github/workflows/api-build-lint-push-containers.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and push container image (latest)
+        # Uncomment the following line for testing
```
Comment (jfagoagas) [inline .github/workflows/api-build-lint-push-containers.yml]:
```suggestion
        # Comment the following line for testing
```

**Review by drewkerrigan — APPROVED** [review]
🚀

**Review by drewkerrigan — APPROVED** [review]

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**